README.md

```diff
@@ -1,92 +1,170 @@
 # WebRTC-grabber
 ---
 
-**Main use case for this project:**  
-Streaming live screen video from contestants' screens on ICPC World Finals 
-as a part of ICPC Live broadcast.
+The main use case is streaming live screen video from contestants' screens on
+[ICPC World Finals](https://icpc.global/) as a part of
+[ICPC Live](https://live.icpc.global/) broadcast.
+
+**Table of Contents**
+
+1. [WebRTC Protocol](#WebRTC-Protocol)
+2. [Grabber](#Grabber)
+    - [Configuration](#grabber-config)
+    - [Build sources](#grabber-build)
+    - [Run](#grabber-run)
+        - [Docker](#grabber-docker)
+        - [Executable (Windows, Linux, MacOS)](#grabber-exec)
+3. [Signaling](#Signaling)
+    - [Configuration](#signaling-config)
+    - [Build sources](#signaling-build)
+    - [Run](#signaling-run)
+        - [Docker](#signaling-docker)
+        - [Executable (Windows, Linux, MacOS)](#signaling-exec)
+4. [TURN](#TURN)
+5. [FAQ](#FAQ)
+6. [License](#License)
+
+## WebRTC Protocol
+---
 
+[WebRTC](https://webrtc.org/) is a new protocol widely used for video
+calling and screen sharing. It works over a local network, allows to start
+the stream on-demand within roughly 1 sec and can be distributed portably.
+There are multiple public implementations of WebRTC protocols, including
+electron applications that we use as a reference point in this project.
 
+<img src="img/webrtc-protocol.png">
 
-## Signalling
----
+WebRTC works by connecting each peer with each other to achieve peer to
+peer communication.
+In our example, peer A would be the team computer, and peer B would be our
+computer for live-streaming (since it’s the one that’s actually showing the
+feed).
 
-A socket.io node js server for signaling between the players and grabbers.
+<img src="img/webrtc-example.png">
 
-Runs on `localhost` port `3000`.
+In the environment of an ICPC competition, it’s hard to allow communication
+between a computer inside the “blue” network and “red” network. WebRTC
+needs ranges of UDP and TCP ports in order to send video. Opening all of
+them to any device in the “red” network didn’t seem plausible. So we
+decided upon using a **TURN** server on the edge of two networks to allow for
+communication between those networks.
+TURN is a server that provides service of connecting computers behind
+multiple **NAT**-s.
 
-Start using (in `packages/relay` directory)
+It is able to do so because it has a constant globally available IP address
+and anybody can connect to it using any TCP/UDP port. Then, after both
+parties established a connection to it, the server passes all the data
+between the peers.
+We decided upon using it because it seemed like it’s the easiest way to
+connect peers from the “blue” network with computers from the “red” network.
 
-```
-npm ci
-npm run start:dev
-```
+The solution consisted of 3 main parts:
 
-or as docker image
+- **Grabber**
```

Comment:
Let describe parts of sevice:

* Grabber - standalone application for background capturing and sharing sceen and webcam
* Signalling - webserver to exchange connection data between the grabber and players, capture video in browser and maintain a list of active grabbers
* Turn - server for forwarding video packages between grabber and players when thay separated usin nat

Reply:
Added